Assets/Scripts/Layers/Adapters/DataTypeAdapters/WMSImportAdapter.cs

```diff
@@ -0,0 +1,363 @@
+                if (sourceUrl.ToLower().Contains("version="))
+                {
+                    wmsVersion = sourceUrl.ToLower().Split("version=")[1].Split("&")[0];
+                    string[] numbers = wmsVersion.Split('.');
+                    int major = int.Parse(numbers[0]);
+                    int minor = int.Parse(numbers[1]);
+                    int revision = int.Parse(numbers[2]);                    
+                    int targetMajor = 1;
+                    int targetMinor = 3;
+                    int targetRevision = 0;
+                    bool isHigherOrEqualVersion = (major > targetMajor) ||
+                                           (major == targetMajor && minor >= targetMinor) ||
+                                           (major == targetMajor && minor == targetMinor && revision >= targetRevision);
+                    coordinateSystemType = isHigherOrEqualVersion ? "CRS" : "SRS";
+                    coordinateSystemReference = defaultCoordinateSystemReference;
+                }
+                else
+                {
+                    Debug.LogWarning("WMS version could not be determined, defaulting to " + defaultFallbackVersion);
+                    wmsVersion = defaultFallbackVersion;
+                    coordinateSystemType = defaultCoordinateSystemType;
+                    coordinateSystemReference = defaultCoordinateSystemReference;
+                }
+
+                wmsParam.spatialReferenceType = coordinateSystemType;
+                wmsParam.spatialReference = coordinateSystemReference;
+                wmsParam.style = GetParamValueFromSourceUrl(sourceUrl, "styles");
+
+                AddWMSLayer(wmsParam, url, wmsFolder, 0 < layerPrefab.DefaultEnabledLayersMax);
+
+                wms = null;
+                return;
+            }
+        }
+
+        private string GetParamValueFromSourceUrl(string sourceUrl, string param)
+        {
+            string value = string.Empty;
+            string p = param + "=";
+            if (sourceUrl.ToLower().Contains(p))
+            {
+                value = sourceUrl.ToLower().Split(p)[1].Split("&")[0];
+            }
+            return value;
+        }
+
+        private void AddWMSLayer(WMS.WMSLayerQueryParams layer, string sourceUrl, FolderLayer folderLayer, bool defaultEnabled)
+        {
+            //Spawn a new WMS layer
+            WMSLayerGameObject newLayer = Instantiate(layerPrefab);
+            newLayer.LayerData.SetParent(folderLayer);
+            newLayer.Name = layer.name;            
+
+            Debug.Log("Adding WMS layer: " + layer.name);
+            // Create a layerType URL for the specific layerType
+            UriBuilder uriBuilder = CreateLayerUri(layer, sourceUrl);
+            var getLayerTypeUrl = uriBuilder.Uri.ToString();
+            string finalUrl = Uri.UnescapeDataString(getLayerTypeUrl);            
+            
+            newLayer.SetURL(finalUrl);
```

Comment:
In plaats van SetURI kan je beter hier het property object aanmaken met de URI erin, en die setten. Dan heb je namelijk maar 1 source of truth, en dat is dat je property object de URI heeft. Nu is het toewijzen van de URI verspreid over deze methode, en loadproperties; terwijl het toch in die property moet zijn